engine/src/main/java/org/terasology/persistence/typeHandling/coreTypes/factories/ListTypeHandlerFactory.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.persistence.typeHandling.coreTypes.factories;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.persistence.typeHandling.TypeHandler;
+import org.terasology.persistence.typeHandling.TypeHandlerFactory;
+import org.terasology.persistence.typeHandling.TypeSerializationLibrary;
+import org.terasology.persistence.typeHandling.coreTypes.ListTypeHandler;
+import org.terasology.reflection.TypeInfo;
+import org.terasology.utilities.ReflectionUtil;
+
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Creates a {@link ListTypeHandler} for {@link List} types.
+ */
+public class ListTypeHandlerFactory implements TypeHandlerFactory {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ListTypeHandlerFactory.class);
```

Comment:
`LOGGER` is traditionally just `logger` for some reason even though it seems like it would be a style violation

Reply:
Shouldn't we be following the style guide? `LOGGER` is a constant (`static final`) field and should be named in `SCREAMING_SNAKE_CASE` according to the style guide.

If you do want me to make the change, do you want it now or can it be done later? I anyway plan on replacing this `TypeHandlerFactory` with a more general `CollectionTypeHandlerFactory` later on in this PR and can make the change then.